### PR TITLE
[cli][build-utils] Use `handle: error` for api 404

### PR DESCRIPTION
**Pull request by styfle**
This PR does a few things:

- Consolidates build-utils to use both 404 routes in `handle: error` phase
- Fixes `vc dev` to process `handle: error` phase after the `handle: filesystem` phase
- Updates `/api` 404 route to be excluded for frameworks like RedwoodJS which use their own functions

### CODE REVIEW

File: packages/now-build-utils/src/detect-builders.ts [hunk 1]
```diff
@@ -971,10 +971,10 @@ function getRouteResult(
...
 
       rewriteRoutes.push(...dynamicRoutes);
-      rewriteRoutes.push({
-        src: '^/api(/.*)?$',
+      errorRoutes.push({
         status: 404,
-        continue: true,
+        src: '^/api(/.*)?$',
+        dest: '/_vc_no_match_',
       });
```
Comment (styfle) [inline packages/now-build-utils/src/detect-builders.ts]: @dav-is What are your thoughts on having two `handle: error` routes?

Can we handled this correctly in production today and use the default 404 page for `/api` and the custom `/404.html` page for all others?

```js
  [
      {
        status: 404,
        src: '^/api(/.*)?$',
        dest: '/_vc_no_match_',
      },
      {
        status: 404,
        src: '^/(?!.*api).*$',
        dest: '/404.html',
      },
  ]
```

Comment (dav-is) [reply packages/now-build-utils/src/detect-builders.ts]: What is `/_vc_no_match`? You should be able to just leave off the `dest`

Comment (dav-is) [reply packages/now-build-utils/src/detect-builders.ts]: `^/(?!.*api).*$` also matches `/docs/api` is that intended?

Comment (styfle) [reply packages/now-build-utils/src/detect-builders.ts]: I thought we had to use `dest` so I used something that wouldn't match: `_vc_no_match`

I'll see if it works without the `dest`.

The two `src` matches here are supposed to be opposites: one matches `/api` and paths below, and the other matches everything besides `/api`. So yes, it is intended.

Comment (dav-is) [reply packages/now-build-utils/src/detect-builders.ts]: `dest` is only required when changing the path. If it's in handle: error. It must not exist, there's no point in adding another filesystem check

How is `/test/api` and `/api` opposites? It's not in the API directory

Comment (dav-is) [reply packages/now-build-utils/src/detect-builders.ts]: Oh sorry I meant to say, it **doesn't** match `/docs/api`, so it wouldn't see the custom 404 page

Comment (styfle) [reply packages/now-build-utils/src/detect-builders.ts]: > Oh sorry I meant to say, it doesn't match /docs/api, so it wouldn't see the custom 404 page

Oh yes that is a problem then. Maybe the solution is to use a wildcard to match anything and the order should make it work correctly?

```js
  [
      {
        status: 404,
        src: '^/api(/.*)?$',
      },
      {
        status: 404,
        src: '^/.*$',
        dest: '/404.html',
      },
  ]
```

*Updated* to fix typo

Comment (dav-is) [reply packages/now-build-utils/src/detect-builders.ts]: I think that'd work

Comment (dav-is) [reply packages/now-build-utils/src/detect-builders.ts]: Remember the `.html` is conditional on `cleanUrls`

Comment (dav-is) [reply packages/now-build-utils/src/detect-builders.ts]: Why is it `^/(.*)*$` and not `^/.*$`?

**Review by AndyBitz — DISMISSED** [review]

**Review by AndyBitz — APPROVED** [review]

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**